### PR TITLE
fix(jira): change source to connection

### LINKED ISSUE #1780: Rename `jira_source` to `jira_connection`
## Description
With respect to #1770, `jira_source` is not a suitable term for what it represents. It is quite confusing from our source code:

image

## Describe the solution you'd like
1. Rename `jira_source` to `jira_connection`
2. Combine `jira_source` and `jira_connection` api
3. Change api path `sources` to `connections` for all plugins

## Has the Feature been Requested Before?
No

### PR DESCRIPTION
**Pull request by warren830**
# Summary

1. For jira, changed all sourceId, jira_sources to connectionId, jira_connections
2. For other plugins, changed api path from /sources to /connections
3. Also changed source in front end

### Key Points

- [ ] This is a breaking change
- [ ] New or existing documentation is updated

### Description
Describe what this PR does, and aims to solve in a few sentences.

### Does this close any open issues?
close #1780

### Current Behavior
Describe the current behaviour of this issue, if relevant.

### New Behavior
Describe the new behaviour updated in this issue, if relevant.

### Screenshots
.
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/ae/api/source.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ type AEResponse struct {
 }
 
 /*
-PUT /plugins/ae/sources/:sourceId
+PUT /plugins/ae/connections/:sourceId
```
Comment (klesh) [inline plugins/ae/api/source.go:27]: `sourceId` should be updated as well

File: plugins/ae/api/source.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func ListSources(input *core.ApiResourceInput) (*core.ApiResourceOutput, error)
 }
 
 /*
-GET /plugins/ae/sources/:sourceId
+GET /plugins/ae/connections/:sourceId
```
Comment (klesh) [inline plugins/ae/api/source.go:72]: same as above

File: plugins/github/api/source.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ type GithubResponse struct {
 }
```
Comment (klesh) [inline plugins/github/api/source.go:31]: file name should updated as well

File: plugins/gitlab/gitlab.go [hunk 4]
```diff
@@ -89,11 +89,11 @@ func (plugin Gitlab) ApiResources() map[string]map[string]core.ApiResourceHandle
...
 			"POST": api.TestConnection,
 		},
-		"sources": {
+		"connections": {
 			"GET":  api.ListSources,
 			"POST": api.PutSource,
 		},
-		"sources/:sourceId": {
+		"connections/:sourceId": {
 			"GET": api.GetSource,
```
Comment (klesh) [inline plugins/gitlab/gitlab.go:97]: method names should be updated as well

File: plugins/github/api/source.go (hunk 3 above)
Comment (warren830) [reply plugins/github/api/source.go:31]: Will update in the new pr

File: plugins/gitlab/gitlab.go (hunk 4 above)
Comment (warren830) [reply plugins/gitlab/gitlab.go:97]: will update in the new pr

**Review by klesh — APPROVED** [review]
LGTM

**Comment by klesh** [comment]
@e2corporation Hi, Julien, please note that `source` has been renamed to `connection` to avoid term-confusion. `config-ui` is also updated by this PR, you may want to take a look.

**Pull request merged**